pandas/core/window.py

```diff
@@ -623,20 +624,30 @@ def _validate_win_type(win_type, kwargs):
                 arg_map = {'kaiser': ['beta'],
                            'gaussian': ['std'],
                            'general_gaussian': ['power', 'width'],
-                           'slepian': ['width']}
+                           'slepian': ['width'],
+                           'exponential': ['center', 'tau']}
+                immutable_args_map = {'exponential': {'center': None}}
```

Comment:
this map is here because ```window``` param in ```get_window``` function (https://docs.scipy.org/doc/scipy-0.19.1/reference/generated/scipy.signal.get_window.html) for exponential window (https://docs.scipy.org/doc/scipy-0.19.1/reference/generated/scipy.signal.exponential.html#scipy.signal.exponential) needs parameters ```center``` and ```tau```. But we do not want to set ```center``` here.

Reply:
Not sure I completely follow - do you have a test case that is supposed to hit this?

Reply:
I will try to describe it more deeply. Maybe we will find a better solution.

Window is created by calling ```sig.get_window``` (https://github.com/pandas-dev/pandas/blob/3876dd8db569ef9de673cbb758e93a76e03b4b08/pandas/core/window.py#L655). For this function, you have to prepare tuple ```win_type``` that consists of few items (win_name, win_param_1, win_param_2, ...) - it was already implemented and works perfectly for previously supproted windows. 
For example for gaussinan window, you have to call scipy function ```sig.get_window``` like this ```w = sig.get_window(("gaussian", 1), N, True)```. The tuple ```("gaussian", 1)``` is inside ```scipy.singal``` parsed and appropriate function is then called e.g. here for gaussian is called ```scipy.signal.gaussian(M, std, sym=True)``` -> ```scipy.signal.gaussian(N, 1)```

The exponential window is defined like this ```scipy.signal.exponential(M, center=None, tau=1.0, sym=True)```, and the problem is in ```center``` parameter. I need to prepare ```win_type``` (the tuple) that contains ```None``` as the first window parameter, but is not user parameter passed in kwargs (and should not be). So for exponential window, I need the tuple like this ('exponential', None, tau). This is the main complexity here that I need to add aditional param that is not passed by user in kwargs.

About the test. The ```immutable_args_map``` functionality is tested when ```test_cmov_window_special``` is called for exponential window.

Reply:
Hmm...it feels somewhat special-case-y, but this seems to be borne out of `scipy` API.  I don't feel great about this, but I do understand the necessity.

Reply:
I checked all possible windows in scipy once again and I did not find another window, where we would use this approach. For other windows (if they will be in future implemented), the parameters exposed in pandas will be the same as the ones passed to underlying scipy function ```sig.get_window()```. So maybe this ```immutable_args_map``` is overkill here. I will try to come another (more lightweight) solution.